pkg/devcontainer/mocks/mock_docker_client.go

```diff
@@ -0,0 +1,56 @@
+package mocks
+
+import (
+	"context"
+	"io"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/network"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// MockDockerClient is a mock implementation of the DockerClient interface for testing
+type MockDockerClient struct {
```

Comment:
Lets try [this](https://github.com/samalba/dockerclient/blob/master/mockclient/mock.go#L10)

Reply:
Do or do not. There is no try ©

Reply:
Done